server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java

```diff
@@ -561,7 +563,18 @@ private Entry(StreamInput in) throws IOException {
             includeGlobalState = in.readBoolean();
             partial = in.readBoolean();
             state = State.fromValue(in.readByte());
-            indices = in.readList(IndexId::new);
+            final int indexCount = in.readVInt();
+            if (indexCount == 0) {
+                indices = Collections.emptyMap();
+            } else {
+                final Map<String, IndexId> idx = new HashMap<>(indexCount);
+                for (int i = 0; i < indexCount; i++) {
+                    final String name = in.readString();
```

Comment:
Can we use `new IndexId(in)` instead, to ensure any wire protocol changes to `IndexId` are centralized?

Reply:
++